QuickDeployment/QuickDeploymentInternetSetup.md

```diff
@@ -0,0 +1,215 @@
+
+1. In the Roles section, double-click the `chocorole` Role to to add it to the `Granted` section. Click the `Create local user` button to create your user.
+
+1. Under the `Security` sidebar menu, select `Anonymous Access`. Un-check the box next to the option `Allow anonymous users to access the server`, and click `Save`.
+
+1. On your endpoints, you can now set up your internal source repository using this newly created `chocouser` credential. The below command is an example; please adjust according to your FQDN, repository name, and user credential created:
+
+    ```powershell
+    choco source add -n "'ChocolateyInternal'" -s "'https://chocoserver:8443/repository/ChocolateyInternal/'" --user='chocouser' --password='YOUR_PASSWORD' --allow-self-service
+    ```
+
+> :memo: **Note:**
+> Now that you've added a credential to your Nexus repositories, access to the `ChocolateyInstall.ps1` and `ClientSetup.ps1` scripts in your `choco-install` raw repository will require this credential as well. 
+
+As we will learn in the next section on CCM, there will be more changes we need to incorporate into these two scripts. Script adjustments are discussed further in the [Adjusting Scripts for Client Setup](#adjusting-scripts-for-client-setup) section below.
+
+## CCM Setup
+
+QDE V1 does not currently include the most up-to-date version of the CCM packages (version 0.3.0, as of this writing). If you have already purchased your Chocolatey for Business (C4B) licenses, you can upgrade by following the [[Central Management Upgrade|CentralManagementSetupUpgrade]] documentation. If you are a trial user, please reach out to your Sales representative for the appropriate packages and procedure for upgrading CCM.
+
+An additional mechanism of security that is highly recommended is the addition of salt additives to the encrypted communication between your endpoints and the CCM Service. As we know, communication from the endpoints to CCM occurs over port 24020, and is secured by SSL certificates. Adding a salt additive on both ends further hashes this encrypted data, and provides another layer of verification. These salt additives should be at least 8 characters, and you will need to provide both additives on the CCM server and endpoint ends when setting up your communication. These are both configuration items that can be set using the `choco config` command, as shown in the example here:
+
+```powershell
+choco config set centralManagementClientCommunicationSaltAdditivePassword 'YourSuperSecureSalt1'
+choco config set centralManagementServerCommunicationSaltAdditivePassword 'YourSuperSecureSalt2'
+```
+
+Further details on configuring CCM, and all available settings, can be found in the [[Central Management Client Setup|CentralManagementSetupClient#config-settings]] documentation.
+
+In the next section, you will need to incorporate both these salt additives into the script that helps you set up your endpoint clients.
+
+## Adjusting Scripts for Client Setup
```

Comment:
Does this edit the hosts file? I didn't see talk of that, and with self-signed it is quite likely that one will need to edit the hosts file.

Reply:
This script does not. I have included mention of the HOSTS file requirement in the Self-Signed scenario section.

Reply:
It may or may not just be in that scenario, but it's good to add for sure.

Reply:
If you are using a Domain certificate, the assumptions is you control your domain's DNS records, and can create a record for this FQDN resolving to the QDE IP (as mentioned in that section when we say to ensure that resolution exists).

If you are using purchased/acquired CA, again, as long as the FQDN of the "Subject/CommonName" resolves to the QDE IP, you're good.

IMHO, editing HOSTS file is more of a work-around, and that record does not cleanup or update if you change your environment/topology.

Reply:
I have included the following warning at the beginning of this document:

> :closed_lock_with_key:**WARNING:** This document assumes you have an understanding of the underlying principles behind **SSL Certificates** and **Public Key Infrastructure**. Therefore, it does not exhaustively cover these topics, but rather presents steps and examples of what you will need to accomplish. You should be able to manage certificates in your organizational environment before proceeding any further.